### PR TITLE
New text inputs

### PR DESCRIPTION
**Pull request by samholmes**
### CHANGELOG

Does this branch warrant an entry to the CHANGELOG?

- [x] Yes
- [ ] No

### Dependencies

 none

### Description

 none

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206275699598774

### CODE REVIEW

File: src/components/modals/ListModal.tsx [hunk 1]
```diff
@@ -70,7 +72,19 @@ export function ListModal<T>({
...
+  autoCorrect = false,
+  autoCapitalize = 'words',
+  returnKeyType = 'search',
+  keyboardType,
+  blurOnSubmit,
+  inputAccessoryViewID,
+  maxLength,
+  secureTextEntry,
+  autoFocus,
+  blurOnClear = true
```
Comment (paullinator) [inline src/components/modals/ListModal.tsx:85]: Blur on clear is actually a very bad UX. I know it's not a part of this task, but please default to `false`

File: src/components/layout/Space.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import * as React from 'react'
```
Comment (paullinator) [inline src/components/layout/Space.tsx:1]: Due to now known performance impact of additional `View` components, I would push back against the architecture of the `Space` component and to use padding on existing components instead where possible.

File: src/components/scenes/ChangePasswordScene.tsx [hunk 3]
```diff
@@ -121,34 +122,34 @@ const ChangePasswordSceneComponent = ({
...
-            label={lstrings.password}
-            autoFocus={focusFirst}
-            onChangeText={handleValidatePassword}
-            onSubmitEditing={handleFocusSwitch}
-            clearIcon
-            searchIcon={false}
-            marginRem={[0, 0.75, 1.25]}
-            maxLength={100}
-          />
+          <Space horizontal={0.75} bottom={1.25}>
```
Comment (paullinator) [inline src/components/scenes/ChangePasswordScene.tsx:125]: Due to now known performance impact of additional `View` components, I would push back against the architecture of the `Space` component and to use padding on existing components instead where possible.

**Review by paullinator — APPROVED** [review]

**Review by paullinator — CHANGES_REQUESTED** [review]
Look good. All comments are optional and doable on a separate PR. FilledTextInput still needs spacing fixes just like in the GUI so I'd ask for a fixup commit to adjust the spacing before merging so we don't need the bureaucracy of another task/PR

File: src/components/scenes/ChangePasswordScene.tsx (hunk 3 above)
Comment (samholmes) [reply src/components/scenes/ChangePasswordScene.tsx:125]: What known performance impact of additional view components?

File: src/components/themed/SimpleTextInput.tsx [hunk 4]
```diff
@@ -332,6 +339,50 @@ const SideContainer = styled(Animated.View)<{ size: SharedValue<number> }>(
...
+      position: 'absolute',
+      top: 0,
+      alignItems: 'center',
+      justifyContent: 'center',
+      paddingHorizontal: theme.rem(0.5),
+      paddingVertical: 0,
+      margin: 0
+    },
+    {
+      color: theme.primaryText,
```
Comment (paullinator) [inline src/components/themed/SimpleTextInput.tsx:369]: This should use `theme.textInputTextColor`

**Review by paullinator — CHANGES_REQUESTED** [review]
Only one minor comment but it's worth fixing. Also can you provide screenshots for visual review?

**Review by paullinator — APPROVED** [review]

**Pull request merged**